tests/console/journalctl.pm

```diff
@@ -23,45 +23,45 @@
 #             Sergio Lindo Mansilla <slindomansilla@suse.com>
 # Tags: bsc#1063066 bsc#1171858
 
-use base "consoletest";
-use strict;
-use warnings;
-use Date::Parse;
+use Mojo::Base qw(consoletest);
+use Date::Parse qw(str2time);
 use testapi;
-use utils;
-use version_utils;
-use power_action_utils 'power_action';
+use utils qw(zypper_call script_retry systemctl);
+use version_utils qw(is_opensuse is_tumbleweed is_sle is_public_cloud);
+use Utils::Backends qw(is_hyperv);
+use power_action_utils qw(power_action);
+use constant {
+    PERSISTENT_LOG_DIR => '/var/log/journal',
+    DROPIN_DIR         => '/etc/systemd/journald.conf.d/',
+    SYSLOG             => '/var/log/messages'
+};
 
-sub check_journal {
+sub is_journal_empty {
     my ($args, $filename) = @_;
     assert_script_run("journalctl -q $args > $filename");
-    return script_run("if [ -s $filename ]; then true; else false; fi");
-}
-
-sub skip_fss_check {
-    # FSS check is not supported by SLE. Disable this check for SLES
-    return is_sle;
-}
-
-sub check_syslog {
-    # rsyslog is not installed on tumbleweed anymore
-    return !is_tumbleweed && !is_jeos;
+    return (script_run("[ -s $filename ]") != 0);
 }
 
 sub verify_journal {
+    my $fss_key = shift;
     # Run journalctl --verify and on failure check for 'File corruption detected'
     # if that happens, run it again after waiting some time and softfailure to https://bugzilla.suse.com/show_bug.cgi?id=1178193
-    my $cmd = "journalctl --verify";
-    $cmd = 'journalctl --verify --verify-key=`cat /var/tmp/journalctl-setup-keys.txt`' unless skip_fss_check;
+    my $cmd = 'journalctl --verify';
+    $cmd .= " --verify-key=$fss_key" if defined($fss_key);
+
+    #No sealing yet
 
     return if (script_run("$cmd 2>&1 | tee errs") == 0);
     # Check for https://bugzilla.suse.com/show_bug.cgi?id=1171858, corruption bug when FSS is enabled
-    if (!skip_fss_check && script_run("grep 'tag/entry realtime timestamp out of synchronization' errs") == 0) {
-        record_soft_failure "bsc#1171858";
+    if (defined($fss_key) && (script_run("grep 'tag/entry realtime timestamp out of synchronization' errs") == 0)) {
+        # upstream issue
+        # https://github.com/systemd/systemd/issues/17833
+        record_soft_failure 'bsc#1171858 - journal corruption: "tag/entry realtime timestamp out of synchronization"';
         # Check for https://bugzilla.suse.com/show_bug.cgi?id=1178193, a race condition for `journalctl --verify`
```

Comment:
it seems to me that this comment now pointing to wrong line

Reply:
Move to https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/8c17709c476501c91a86c2189905fd977475d623/tests/console/journalctl.pm#L62